lib/ansible/modules/cloud/amazon/rds_instance.py

```diff
@@ -0,0 +1,1191 @@
+'''
+
+RETURN = '''
+instance:
+  description:
+    - the information returned in data from boto3 get_db_instance or from modify_db_instance
+      converted from a CamelCase dictionary into a snake_case dictionary
+  returned: success
+  type: dict
+changed:
+  description:
+    - whether the RDS instance configuration has been changed.  Please see the main module
+      description.  Changes may be delayed so, unless the correct parameters are given
+      this does not mean that the changed configuration has already been implemented.
+  returned: success
+  type: bool
+response:
+  description:
+    - the raw response from the last call to AWS if available.  This will likely include
+      the configuration of the RDS in CamelCase if needed
+  returned: when available
+  type: dict
+'''
+
+from ansible.module_utils.six import print_
+import sys
+import time
+import traceback
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import get_aws_connection_info, boto3_conn
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict, AWSRetry
+from ansible.module_utils.ec2 import ansible_dict_to_boto3_tag_list, boto3_tag_list_to_ansible_dict, compare_aws_tags
+from ansible.module_utils.aws.rds import get_db_instance, instance_to_facts, instance_facts_diff
+from ansible.module_utils.aws.rds import DEFAULT_PORTS, DB_ENGINES, LICENSE_MODELS
+try:
+    import botocore
+    from botocore import xform_name
+except ImportError:
+    pass  # caught by imported AnsibleAWSModule
+
+
+# NOTE ON DATA STRUCTURES AND NAMING:
+
+# We use raw dictionaries in two variants as our main data structures.  One format is the
+# snake_case formatted facts dictionaries whilst the other is the CamelCased AWS parameter
+# dictionaries.  We try to keep data for as long as reasonable in the format it arrived
+# in.
+#
+#   - Generally AWS CamelCased formatted dictionaries of instance data are named instance*
+#   - Generally Ansible snake_case formatted dictionaries of facts about instaces are named fact*
+#
+# the one exception is that our final return is in fact format but the return parameter
+# with the instance facts is called "instance".
+#
+# Dicts are chosen rather than than objects becuase
+#   - they match with use elsewhere (AWS returns a dict, facts are passed in dicts)
+#   - they are somewhat simpler to manipluate in many cases
+#
+
+
+# Q is a simple logging framework NOT suitable for production use but handy in development.
+# try:
+#     import q
+#     HAS_Q = True
+# except:
+#     HAS_Q = False
+
+HAS_Q = False
+
+
+def q(junk):
+    pass
+
+
+# Ansible avoids proper python logging for reasons too complicted for
+# me to fully understand but including the need to properly handle
+# security (see proposals, many tickets and a number of PRs), so we
+# make a little improvised logger here which goes through module.log
+# we can always point to a more sophisticated logger when one exists.
```

Comment:
Is the purpose of this logger just to shim for `module.log` and ship to `q` when that's installed?